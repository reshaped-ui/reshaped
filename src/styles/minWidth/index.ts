import { responsiveClassNames, responsiveVariables } from "utilities/helpers";
import * as T from "styles/types";
import s from "./minWidth.module.css";

const getMinWidthStyles: T.DynamicStyleUtility<string | number> = (value) => {
	if (!value) return null;
	const variables = responsiveVariables("--rs-min-w", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "root-unit" : "root-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames, variables };
};

export default getMinWidthStyles;
