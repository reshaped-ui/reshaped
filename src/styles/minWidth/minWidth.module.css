.literal {
	min-width: var(--rs-min-w-s, auto);
}

.unit {
	min-width: calc(var(--rs-min-w-s) * var(--rs-unit-x1));
}

@each $viewport in m, l, xl {
	@media (--rs-viewport-$viewport) {
		.literal--$viewport {
			min-width: var(--rs-min-w-$viewport);
		}

		.unit--$viewport {
			min-width: calc(var(--rs-min-w-$viewport) * var(--rs-unit-x1));
		}
	}
}
