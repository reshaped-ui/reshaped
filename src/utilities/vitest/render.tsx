import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
