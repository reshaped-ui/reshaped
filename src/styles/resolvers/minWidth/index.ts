import { responsiveClassNames, responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import s from "./minWidth.module.css";

const getMinWidthStyles: T.StyleResolver<T.MinWidth> = (value) => {
	if (!value) return {};
	const variables = responsiveVariables("--rs-min-w", value);
	const classNames = responsiveClassNames(
		s,
		(value) => (typeof value === "number" ? "--type-unit" : "--type-literal"),
		value,
		{ excludeValueFromClassName: true }
	);

	return { classNames: [s.root, classNames], variables };
};

export default getMinWidthStyles;
