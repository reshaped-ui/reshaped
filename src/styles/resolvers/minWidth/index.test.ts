import { expect, test, describe } from "vitest";
import minWidth from "./index";

describe("Styles/MinWidth", () => {
	test("handles px value", () => {
		expect(minWidth("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(minWidth("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(minWidth(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(minWidth()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(minWidth({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
