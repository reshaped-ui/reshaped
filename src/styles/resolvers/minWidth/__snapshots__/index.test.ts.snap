// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MinWidth > handles % value 1`] = `
{
  "classNames": [
    "_root_f00fd4",
    [
      "_--type-literal_f00fd4",
    ],
  ],
  "variables": {
    "--rs-min-w-s": "50%",
  },
}
`;

exports[`Styles/MinWidth > handles px value 1`] = `
{
  "classNames": [
    "_root_f00fd4",
    [
      "_--type-literal_f00fd4",
    ],
  ],
  "variables": {
    "--rs-min-w-s": "50px",
  },
}
`;

exports[`Styles/MinWidth > handles responsive value 1`] = `
{
  "classNames": [
    "_root_f00fd4",
    [
      "_--type-literal_f00fd4",
      "_--type-literal--l_f00fd4",
    ],
  ],
  "variables": {
    "--rs-min-w-l": "50%",
    "--rs-min-w-s": "50px",
  },
}
`;

exports[`Styles/MinWidth > handles undefined value 1`] = `{}`;

exports[`Styles/MinWidth > handles unit value 1`] = `
{
  "classNames": [
    "_root_f00fd4",
    [
      "_--type-unit_f00fd4",
    ],
  ],
  "variables": {
    "--rs-min-w-s": 5,
  },
}
`;
